+++ src/lib/libpcap/pcap-linux.c
@@ -129,11 +129,11 @@
 #include <sys/ioctl.h>
 #include <sys/utsname.h>
 #include <sys/mman.h>
-#include <linux/if.h>
-#include <linux/if_packet.h>
+#include <net/if.h>
+#include <net/if_packet.h>
 #include <linux/sockios.h>
 #include <netinet/in.h>
-#include <linux/if_ether.h>
+#include <net/if_ether.h>
 #include <net/if_arp.h>
 #include <poll.h>
 #include <dirent.h>
